### ISSUE TITLE
Retire cf-custom-resources/lib/desired-count-delegation.js

### ISSUE THREAD

**Issue by miketheman** [aws/copilot-cli#4442]
This custom resource was created prior to CloudFormation making the DesiredCount value optional.

> -- DesiredCount becomes an optional field in CFN CreateService and DesiredCount=1 would be used as the default value if it is missing
> -- UpdateService will also omit DesiredCount when it is missing in CFN template

Refs: https://github.com/aws/containers-roadmap/issues/493#issuecomment-721421554

Would cut down on the custom resources deployed for a customer, API calls necessary to set the service correctly, and overall complexity.

**Comment by KollaAdithya** [comment]
Hello @miketheman !

Here are the thoughts from @iamhopaul123  on still using the lambda function in Copilot.

The downside of removing the custom resource (desired-count-delegation.js) and to omit the `DesiredCount` in CloudFormation template is discussed in the below scenario.

Suppose the manifest contains and runs `copilot svc deploy` 
```yaml
count   
  range: 5-10 
```

- ECS service will first launch with 1 task(default) and then scale to 5 because of the AutoScalingGroup.  

- The worse scenario for removing the lambda function is that users will get a `slow start ` when they first create the service if they set the minimum capacity to be any non 1 value.

What are your thoughts 🤔 on this?

**Comment by miketheman** [comment]
Hi @KollaAdithya and @iamhopaul123 !

Thanks for taking a look.

(Let's also overlook [the wrong function docstring](https://github.com/aws/copilot-cli/blob/1965fc530210d949de1322ecf27f606b0ff2a9cb/cf-custom-resources/lib/desired-count-delegation.js#L10-L12) 😉 )

From an end-user experience, I agree - specifying that you have a range starting with 5 should mean that you get 5 to begin with - no dispute there.
But once launched, the control over the count is then handled by Autoscaling Policies, and [during deployments](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-auto-scaling.html#auto-scaling-concepts):

> Application Auto Scaling turns off scale-in processes while Amazon ECS deployments are in progress

So in my mind, ECS should already be handling changes based on scale to the `DesiredCount` value, so we should leave it alone.

In regards to:

> The worse scenario for removing the lambda function is that users will get a `slow start ` when they first create the service if they set the minimum capacity to be any non 1 value.

Doesn't that support the case for setting `DesiredCount` only on service create, and never use it after? If that's true, then an adjusting Lambda function seems unnecessary.

**Comment by iamhopaul123** [comment]
> Doesn't that support the case for setting DesiredCount only on service create, and never use it after? If that's true, then an adjusting Lambda function seems unnecessary.

Agreed, The only difficulty here is to have an extra step checking if the stack is already there or not and we'll set `DesiredCount` accordingly (in this case, we only want to set `DesiredCount` if the stack/service is not created).

**Comment by miketheman** [comment]
Sounds like a reasonable check to add, especially if it drops the complexity of a secondary out of band Lambda Function.
